Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
-one row, with one column.
-Now, let us insert three more rows of data,
-using :meth:`~Cursor.executemany`::
+   data = [
+       ("Monty Python Live at the Hollywood Bowl", 1982, 7.9),
+       ("Monty Python's The Meaning of Life", 1983, 7.5),
+       ("Monty Python's Life of Brian", 1979, 8.0),
+   ]
+   cur.executemany("INSERT INTO movie VALUES(?, ?, ?)", data)
+   con.commit()  # Remember to commit the transaction after executing INSERT.
 
-   >>> data = [
-   ...    ('2006-03-28', 'BUY', 'IBM', 1000, 45.0),
-   ...    ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0),
-   ...    ('2006-04-06', 'SELL', 'IBM', 500, 53.0),
-   ... ]
-   >>> cur.executemany('INSERT INTO stocks VALUES(?, ?, ?, ?, ?)', data)
-
-Notice that we used ``?`` placeholders to bind *data* to the query.
+Notice that ``?`` placeholders are used to bind ``data`` to the query.
 Always use placeholders instead of :ref:`string formatting <tut-formatting>`
 to bind Python values to SQL statements,
-to avoid `SQL injection attacks`_.
-See the :ref:`placeholders how-to <sqlite3-placeholders>` for more details.
+to avoid `SQL injection attacks`_
+(see :ref:`sqlite3-placeholders` for more details).
 
-Then, retrieve the data by iterating over the result of a ``SELECT`` statement::
+You can verify that the new rows were inserted
+by executing a ``SELECT`` query,
+this time iterating over the results of the query::
 
-   >>> for row in cur.execute('SELECT * FROM stocks ORDER BY price'):
+   >>> for row in cur.execute("SELECT year, title FROM movie ORDER BY year"):
    ...     print(row)
+   (1971, "And Now for Something Completely Different")
+   (1975, "Monty Python and the Holy Grail")
+   (1979, "Monty Python's Life of Brian")
+   (1982, "Monty Python Live at the Hollywood Bowl")
+   (1983, "Monty Python's The Meaning of Life")
+
+Each row is now a two-item :class:`tuple` of ``(year, title)``.
+
+At last, you'll verify that the database has been written to disk by
+calling :meth:`con.close() <Connection.close>`
+to close the existing connection,
+opening a new one, creating a new cursor,
+then reusing the query from above to read from the database::
+
+   >>> con.close()
+   >>> con2 = sqlite3.connect("tutorial.db")
+   >>> cur2 = con2.cursor()
+   >>> for row in cur.execute("SELECT year, title FROM movie ORDER BY year"):
+   ...     print(row)
+   (1971, "And Now for Something Completely Different")
+   (1975, "Monty Python and the Holy Grail")
+   (1979, "Monty Python's Life of Brian")
+   (1982, "Monty Python Live at the Hollywood Bowl")
+   (1983, "Monty Python's The Meaning of Life")
 
-   ('2006-01-05', 'BUY', 'RHAT', 100, 35.14)
-   ('2006-03-28', 'BUY', 'IBM', 1000, 45.0)
-   ('2006-04-06', 'SELL', 'IBM', 500, 53.0)
-   ('2006-04-05', 'BUY', 'MSFT', 1000, 72.0)
-
-You've now created an SQLite database using the :mod:`!sqlite3` module.
+You've now created an SQLite database using the :mod:`!sqlite3` module,
+inserted data and ran several SQL queries against it.
```

Comment:
Inserting data is technically part of running queries, so maybe you could say something like "...and used its functions and methods to perform some common database operations.".

Reply:
Maybe, though perhaps that's a bit pedantic for a tutorial. Also, what's "its" here? It would be implied to be the database, but I assume you mean `sqlite3` and implicitly its classes?

Reply:
Yes, I meant `sqlite3`.  My suggestion can probably be improved, but the idea was to focus on the fact that the reader should now be able to use the functions/methods provided by the module to interact with the DB and perform queries.

Reply:
I see; in that case, you could say something like

```suggestion
inserted data and performed several common database operations on it.
```

To note, I've framed it in terms of what the reader has _done_, as opposed to what they may have _learned_ to be able to to do. With Diataxis, it is heavily stressed that you should not tell the reader what they've _learned_, as opposed to what they've _done_ (emphasis original):

> **There is no responsibility on the pupil to learn, understand or remember** - the learner’s only obligation in this contract is to do things as directed.

>  Sadly, however much you desire it, you will not be able to learn for your pupil. You cannot make them learn. All you can do is make it so they can learn.

> "*You have built a secure, three-layer hylomorphic stasis engine…*"
>
> Describe (and admire, in a mild way) what your learner has accomplished (note - not: “you have learned…”)

Reply:
I like your latter suggestion (which I have applied, for now), but I'm not totally satisfied with this whole conclusion. I'll revisit it before landing.

Reply:
Yeah, I'm not either, but hopefully you'll have better ideas than I did, heh

Reply:
I updated the final query to show the highest scoring movie, which happens to be _"Monty Python and the Holy Grail"_. We _could_ use the word "grail" in the conclusion 😎...

Reply:
Interesting, when "The Life of Brian" is widely considered to be their most critically acclaimed work...